Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -506,15 +506,16 @@ private static Branch getBranchInstance(@NonNull Context context, boolean isLive
                 branchReferral_.linkCache_.clear();
                 branchReferral_.requestQueue_.clear();
             }
-        }
-        branchReferral_.context_ = context.getApplicationContext();
 
-        /* If {@link Application} is instantiated register for activity life cycle events. */
-        if (context instanceof BranchApp) {
-            isAutoSessionMode_ = true;
-            branchReferral_.setActivityLifeCycleObserver((Application) context);
+            branchReferral_.context_ = context.getApplicationContext();
+            /* If {@link Application} is instantiated register for activity life cycle events. */
+            if (context instanceof BranchApp) {
```

Comment:
Oops - this line: https://github.com/BranchMetrics/android-branch-deep-linking/pull/321/files#diff-11fc8f6026de99b60216e9aca37d6d26L564